lib/client.js

```diff
@@ -618,6 +619,11 @@ class Parser {
       this.connection += buf.toString()
     } else if (key.length === 14 && key.toString().toLowerCase() === 'content-length') {
       this.contentLength += buf.toString()
+    } else if (key.length === 19 && key.toString().toLowerCase() === 'content-disposition') {
+      // https://www.rfc-editor.org/rfc/rfc6266#section-4.3
+      // we need to recieve the buffer as latin1
+      // so it would not transform to utf-8 wrongly
+      this.headers[len - 1] = Buffer.transcode(buf, 'latin1', 'utf8')
```

Comment:
Actually this needs to happen somewhere else.

Reply:
Updated to trans-code inside `onHeadersComplete`